integrated_channels/integrated_channel/exporters/learner_data.py

```diff
@@ -516,39 +557,64 @@ def _collect_certificate_data(self, enterprise_enrollment):
-
         course_id = enterprise_enrollment.course_id
         username = enterprise_enrollment.enterprise_customer_user.user.username
+        user_id = enterprise_enrollment.enterprise_customer_user.user_id
 
-        try:
-            certificate = self.certificates_api.get_course_certificate(course_id, username)
-            completed_date = certificate.get('created_date')
-            if completed_date:
-                completed_date = parse_datetime(completed_date)
-            else:
-                completed_date = timezone.now()
+        completed_date = None
+        grade = self.grade_incomplete
+        is_passing = False
+        percent_grade = None
 
-            # For consistency with _collect_grades_data, we only care about Pass/Fail grades. This could change.
-            is_passing = certificate.get('is_passing')
-            percent_grade = certificate.get('grade')
-            grade = self.grade_passing if is_passing else self.grade_failing
+        try:
+            certificate = get_course_certificate(course_id, username)
+        except InvalidKeyError:
+            self._log_courseid_not_found(course_id, enterprise_enrollment, user_id)
+            return completed_date, grade, is_passing, percent_grade
+
+        if not certificate:
+            self._log_cert_not_found(course_id, enterprise_enrollment, user_id)
+            return completed_date, grade, is_passing, percent_grade
+
+        completed_date = certificate.get('created_date')
+        if completed_date:
+            completed_date = parse_datetime(completed_date)
+        else:
+            completed_date = timezone.now()
 
-        except HttpNotFoundError:
-            LOGGER.error('[Integrated Channel] Certificate data not found.'
-                         ' Course: {course_id}, EnterpriseEnrollment: {enterprise_enrollment},'
-                         ' Username: {username}'.format(
-                             course_id=course_id,
-                             username=username,
-                             enterprise_enrollment=enterprise_enrollment.pk))
-            completed_date = None
-            grade = self.grade_incomplete
-            is_passing = False
-            percent_grade = None
+        # For consistency with _collect_grades_data, we only care about Pass/Fail grades. This could change.
+        is_passing = certificate.get('is_passing')
+        percent_grade = certificate.get('grade')
+        grade = self.grade_passing if is_passing else self.grade_failing
 
         return completed_date, grade, is_passing, percent_grade
 
+    def _log_cert_not_found(self, course_id, enterprise_enrollment, user_id):
+        """
+        Standardized logging for no certificate found (refactor candidate)
+        """
+        LOGGER.error('[Integrated Channel] Certificate not found for user'
+                     ' Course: {course_id}, EnterpriseEnrollment: {enterprise_enrollment}, '
+                     ' Learner id {user_id}'
```

Comment:
Nit: can we label as Learner LMS User Id in the log so it's easy to tell where to go look this up?